rust/src/measurements/make_private_expr/expr_count/mod.rs

```diff
@@ -0,0 +1,145 @@
+    (ExprDomain, PartitionDistance<MI>): MetricSpace,
+{
+    let Expr::Agg(AggExpr::Count(expr, include_nulls)) = expr else {
+        return fallible!(MakeMeasurement, "Expected Count expression");
+    };
+
+    if !include_nulls {
+        return fallible!(
+            MakeMeasurement,
+            "Public margin sizes assume nulls are included. Try using `len` instead."
+        );
+    }
+
+    let t_prior = expr
+        .clone()
+        .make_stable(input_domain.clone(), input_metric.clone())?;
+
+    let (middle_domain, middle_metric) = t_prior.output_space();
+
+    let by = middle_domain.context.grouping_columns()?;
+    let margin = middle_domain
+        .frame_domain
+        .margins
+        .get(&by)
+        .ok_or_else(|| err!(MakeMeasurement, "Unknown margin for {:?}", by))?;
+
+    if Some(MarginPub::Lengths) != margin.public_info {
+        return fallible!(
+            MakeMeasurement,
+            "Len is only private if size(s) of {:?} are public",
+            by
+        );
+    }
+
+    t_prior
+        >> Measurement::new(
+            middle_domain,
+            Function::new_expr(|input_expr| input_expr.len()),
+            middle_metric,
+            MO::default(),
+            PrivacyMap::new(move |_| MO::Distance::zero()),
+        )?
+}
+
+#[cfg(test)]
+mod test_make_expr_count_private {
+    use super::*;
+    use polars::prelude::*;
+
+    use crate::{
+        error::ErrorVariant,
+        measurements::PrivateExpr,
+        measures::MaxDivergence,
+        metrics::{PartitionDistance, SymmetricDistance},
+        transformations::polars_test::get_test_data,
+    };
+
+    #[test]
+    fn test_make_count_expr_grouped() -> Fallible<()> {
+        let (lf_domain, lf) = get_test_data()?;
+        let expr_domain = lf_domain.aggregate(["chunk_2_bool"]);
+        let scale: f64 = f64::NAN;
+
+        let m_lap = col("const_1f64").len().make_private(
+            expr_domain,
+            PartitionDistance(SymmetricDistance),
+            MaxDivergence::default(),
+            scale,
+        )?;
+
+        let meas_res = m_lap.invoke(&(lf.logical_plan.clone(), all()))?;
+
+        let df_actual = lf
+            .clone()
+            .group_by([col("chunk_2_bool")])
+            .agg([meas_res])
+            .collect()?;
+        let df_exact = lf
+            .group_by([col("chunk_2_bool")])
+            .agg([col("const_1f64").len()])
+            .collect()?;
+
+        assert_eq!(
+            df_actual.sort(["chunk_2_bool"], false, false)?,
+            df_exact.sort(["chunk_2_bool"], false, false)?
+        );
+        Ok(())
+    }
+
+    #[test]
+    fn test_make_count_expr_no_length() -> Fallible<()> {
+        let (lf_domain, _) = get_test_data()?;
+        let expr_domain = lf_domain.aggregate(["cycle_5_alpha"]);
+        let scale: f64 = f64::NAN;
+
+        let variant = col("const_1f64")
+            .len()
+            .make_private(
```

Comment:
From the name, I was expecting this not to call `len`, but the difference must be in the data? By itself, `cycle_5_alpha` doesn't really mean much to me.